lisa/wlgen/rta.py

```diff
@@ -2374,11 +2426,12 @@ def __mul__(self, n):
     def to_events(self, **kwargs):
         pass
 
-    def to_json(self, **kwargs):
+    def _to_json(self, plat_info, **kwargs):
         """
         Deduplicate the event names and turn the whole workload into a loop if
         possible.
         """
+        kwargs['plat_info'] = plat_info
```

Comment:
So I faced a similar issue with `to_events()` as with `to_json()`: there isn't a unified signature (not that there should be one). This gets around aligning them.

Reply:
Not sure what you mean exactly, this function does not use plat_info (beyond adding it to its positional argument list), so there should not be any need to mention it in the signature. If the function is simply called with keyword arguments, most `to_json` can just be renamed to `_to_json` without affecting their signature (unless they actually need to use the `plat_info` of course)